app/app.go

```diff
@@ -571,7 +571,23 @@ func wireRecaster(ctx context.Context, eth2Cl eth2wrap.Client, sched core.Schedu
 	broadcaster core.Broadcaster, validators []*manifestpb.Validator, builderAPI bool,
 	callback func(context.Context, core.Duty, core.PubKey, core.SignedData) error,
 ) error {
-	recaster := bcast.NewRecaster()
+	recaster, err := bcast.NewRecaster(func(ctx context.Context) (map[eth2p0.BLSPubKey]struct{}, error) {
+		valList, err := eth2Cl.ActiveValidators(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		ret := make(map[eth2p0.BLSPubKey]struct{})
```

Comment:
nit: can simply have a map[pubkey]bool
```suggestion
		ret := make(map[eth2p0.BLSPubKey]bool)
```

Reply:
using `struct{}` as a dummy value for maps is a well-known practice, i don't think it needs changing